lib/fs.js

```diff
@@ -211,6 +213,7 @@ fs.access = function(path, mode, callback) {
     throw new TypeError('"callback" argument must be a function');
   }
 
+  path = getPathFromURL(path);
```

Comment:
good point. it's not just about null bytes, the path could contain any number of pct-encoded characters. This should run the path through a decode before returning... will fix that